app/views/conversations/show.html.erb

```diff
@@ -48,52 +48,86 @@
   </div>
 
 
-  <div id="right-column" class="flex flex-1 flex-col h-screen bg-white dark:bg-graphite">
+  <div id="right-column" class="flex flex-1 flex-col h-screen bg-white dark:bg-graphite relative">
+
+    <div id="handle-container" class="absolute left-0 top-1/2 w-8 h-20 -mt-10 z-20 hidden md:flex" data-click-toggle-target="trigger">
+      <div id="handle" class="flex-1 flex cursor-pointer group" data-controller="click-toggle" data-click-toggle-target="trigger" data-click-toggle-flippable-class="!hidden">
```

Comment:
I'm confused -- this div has both `data-controller="click-toggle"` and `data-click-toggle-target="trigger"`. Is this the trigger or a new instance of the controller? 

Why use `!hidden` as the toggled class instead of `visible`?

Reply:
I think I understand what's happening now, but I'm still not sure about how it works. It seems like stimulus would get confused. How does it handle nested controllers? 

This is a clever reduction of the patterns of needing to toggle both the sidebar presence and the button states, but I think I'm going to have to reimplement it in the application controller or a dedicated controller to get a keyboard shortcut hooked onto it.

Reply:
Ah, shoot, I didn't consider keyboard shortcuts. But it is possible to add multiple stimulus controllers to a single element, so we might just be able to make a really lightweight "keyboard-to-click" stimulus controller and bind it in the exact same two places this is bound.

This particular line you've tagged is a new instance of the controller, and for this new instance I'm also defining this particular div as the trigger but it doesn't have to be here. It's just a coincidence that the trigger happens to be the same container object which I'm scoping the controller. The CSS class attribute (flippable) must be declared on the same object as the controller (I found that in the reference here: https://stimulus.hotwired.dev/reference/css-classes#attributes).

The click-toggle controller needs two objects: a trigger and a target. The trigger is where it binds the click event and the target is where the class will be toggled.

The reason I toggled `!hidden` is because, unfortunately, the tooltip CSS class is a bit heavy handled and it forces the object to be a block object so I have to override that in order to get the thing to disappear, but even "hidden" isn't enough to override that because the Daisy UI library (where I'm getting the tooltip) is included in application.css *after* tailwinds so it's classes take precedence.

I found guidelines for nested stimulus here: https://stimulus.hotwired.dev/reference/controllers#nested-scopes Basically, it "just works".

Reply:
Wonderful thank you for the links. I'm still learning what Stimulus can do and what it trips on.